tests/integration/s3/test_s3.py

```diff
@@ -3914,6 +3915,12 @@ class TestS3Cors:
             "$..Last-Modified",
         ]
     )
+    @pytest.mark.skip_snapshot_verify(
+        condition=is_old_provider,
+        paths=[
+            "$..x-amz-sdk-checksum-algorithm",
+        ],
+    )
```

Comment:
This indicates that the latest release of moto still contains regression errors (as you referenced, they are addressed in https://github.com/spulec/moto/pull/5765).

Reply:
@alexrashed This is a negative test case (an extra header is returned when it shouldn't be) and unlikely to break user workflows.

I think we can go safely ahead with merging this PR and avoid another iteration with potentially more breakages.

Reply:
Since I cannot really determine the severity of this header being present or not, I would like to get @bentsku's feedback (I added him as a reviewer). I'll follow his judgement, but I would also suggest adding a comment linking the PR and noting that this is just temporary and should be remove with the next upgrade.

Reply:
@bentsku Looping you as the author of the test - please raise any concerns with this change if any.

This change will be reverted with the next moto-ext release.

Reply:
Hi, I need to look deeper, but I don't think this header is ever returned by AWS, even when set in the request. If an algorithm is provided in this request header, the response will return `Checksum{Algorithm}=hash` I think. See https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html

So I suppose this would not break anything client side because it's "just" an extra header never used in response. But it should be fixed in moto and actually never be returned. It should only change the "legacy" provider anyway, as the ASF parser won't pick it up from the moto responses.